### PR TITLE
Introduce fhir-path and fhir-profile modules

### PR DESCRIPTION
**Pull request by JohnTimm**

### CODE REVIEW

File: fhir-path/src/main/java/com/ibm/fhir/path/function/registry/FHIRPathFunctionRegistry.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+import com.ibm.fhir.path.function.ToQuantityFunction;
+import com.ibm.fhir.path.function.ToStringFunction;
+import com.ibm.fhir.path.function.TodayFunction;
+import com.ibm.fhir.path.function.TypeFunction;
+import com.ibm.fhir.path.function.UnionFunction;
+import com.ibm.fhir.path.function.UpperFunction;
+
+public class FHIRPathFunctionRegistry {
+    private static final FHIRPathFunctionRegistry INSTANCE = new FHIRPathFunctionRegistry();
+    private Map<String, FHIRPathFunction> functionMap = new ConcurrentHashMap<>();
```
Comment (prb112) [inline fhir-path/src/main/java/com/ibm/fhir/path/function/registry/FHIRPathFunctionRegistry.java]: small thing... do you intend this to mutable? e.g. a downstream consumer could change this.

File: fhir-validation/pom.xml [hunk 2]
```diff
@@ -18,6 +18,16 @@
...
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>fhir-path</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>fhir-path</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
Comment (prb112) [inline fhir-validation/pom.xml]: is the second one intended for the same scope?

File: fhir-path/src/main/java/com/ibm/fhir/path/function/registry/FHIRPathFunctionRegistry.java (hunk 1 above)
Comment (prb112) [reply fhir-path/src/main/java/com/ibm/fhir/path/function/registry/FHIRPathFunctionRegistry.java]: discussed with John - no direct access

Comment (JohnTimm) [reply fhir-path/src/main/java/com/ibm/fhir/path/function/registry/FHIRPathFunctionRegistry.java]: Added `final` keyword to FHIRPathFunctionRegistry class declaration.

File: fhir-validation/pom.xml (hunk 2 above)
Comment (JohnTimm) [reply fhir-validation/pom.xml]: Removed unnecessary dependency.

**Pull request merged**